src/typeahead/typeahead.ts

```diff
@@ -213,6 +284,29 @@ export class NgbTypeahead implements ControlValueAccessor,
     this._renderer.setProperty(this._elementRef.nativeElement, 'disabled', isDisabled);
   }
 
+  onDocumentClick(event) {
+    if (this.isPopupOpen()) {
+      if (!this._isClickInWidget(event)) {
+        this.dismissPopup();
+      }
+    }
+  }
+
+  _isClickInWidget(event) {
+    const target = event.target;
+
+    const input = this._viewContainerRef.element.nativeElement;
+    return this._isElementOrAncestorOf(input, target);
+  }
+
+  _isElementOrAncestorOf(possibleAncestor, reference) {
```

Comment:
If you think so, I agree with you! It will just make a bigger change regarding the tests (which don't use any DOM content). Also, `util.ts` seems to deal only with primitive values, providing short, simple functions. Does it fit there? Would providing a `dom-util.ts` (or alike) be an alternative?